### PR TITLE
Add understanding results page

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
Add understanding results page to the OpenSearch Benchmark user guide

Addresses part of https://github.com/opensearch-project/documentation-website/issues/5815

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _benchmark/user-guide/understanding-results.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+|                                                 Min Throughput |                                       term |        16.1 |  ops/s |
+|                                                Mean Throughput |                                       term |        16.1 |  ops/s |
+|                                              Median Throughput |                                       term |        16.1 |  ops/s |
+|                                                 Max Throughput |                                       term |        16.1 |  ops/s |
+|                                       100th percentile latency |                                       term |     131.798 |     ms |
+|                                  100th percentile service time |                                       term |     69.5237 |     ms |
+|                                                     error rate |                                       term |           0 |      % |
+```
+
+The unique metrics to yhe cluster begin at the `index` task line. The following two use cases can give you an idea of what metrics might be relevant to you:
```
Comment (IanHoang) [inline _benchmark/user-guide/understanding-results.md:107]: yhe -> the

File: _benchmark/user-guide/understanding-results.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+- To assess how much load your cluster can handle, the `index` task metrics provide the number of documents ingested during the workload run. 
+- To assess how fast you can query against documents in your cluster, the `match_all` and `term` give both the number of query operations performed per second and the measurable latency of the query.
```
Comment (IanHoang) [inline _benchmark/user-guide/understanding-results.md:110]: the measurable latency and service time of the query

Comment (IanHoang) [reply _benchmark/user-guide/understanding-results.md:110]: Maybe add additional point about error rate associated with the ingesting documents and running query operations should be added to both points.

File: _benchmark/user-guide/understanding-results.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+## Result storage
+
+Results from OpenSearch Benchmark are stored in two ways, either in-memory or in an external metric store. 
+
+When stored in-memory, results can be found in the `/.benchmark/benchmarks/test_executions/<test_execution_id>` directory. Results are named based off of the `test_execution_id` given to the workload test during its last run. 
+
+While [running a test](https://opensearch.org/docs/latest/benchmark/reference/commands/execute-test/#general-settings), you can also customize where the results are stored, using any combination of the following command flags:
+
+* `--results-file`: When provided a file path, writes the compare results to the file indicated in the path.
```
Comment (IanHoang) [inline _benchmark/user-guide/understanding-results.md:121]: the compare results --> the summary results

File: _benchmark/user-guide/understanding-results.md (hunk 1 above)
Comment (Naarcha-AWS) [inline _benchmark/user-guide/understanding-results.md:107]:
```suggestion
Metrics unique to the cluster begin at the `index` task line. The following two use cases can give you an idea of what metrics might be relevant to you:
```

File: _benchmark/user-guide/understanding-results.md (hunk 2 above)
Comment (Naarcha-AWS) [inline _benchmark/user-guide/understanding-results.md:110]:
```suggestion
- To assess the measurable latency and service time of the queries in the workload, the `match_all` and `term` give both the number of query operations performed per second and the measurable latency of the query.
```

File: _benchmark/user-guide/understanding-results.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+* `--results-format`: Defines the output format for the command line results, either markdown or csv. Default is `markdown`.
```
Comment (Naarcha-AWS) [inline _benchmark/user-guide/understanding-results.md:122]:
```suggestion
* `--results-format`: Defines the output format for the command line results, either `markdown` or `csv`. Default is `markdown`.
```

File: _benchmark/user-guide/understanding-results.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+* `--show-in-results`: Determines whether or not to include a comparison to previous runs in the results file.
```
Comment (IanHoang) [inline _benchmark/user-guide/understanding-results.md:123]:
```
  --show-in-results {available,all-percentiles,all}
                        Define which values are shown in the summary publish (default: available).
```
Allows users to define which values are shown in the summary results

File: _benchmark/user-guide/understanding-results.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+Results from OpenSearch Benchmark are stored in two ways, either in-memory or in an external metric store. 
+
+When stored in-memory, results can be found in the `/.benchmark/benchmarks/test_executions/<test_execution_id>` directory. Results are named based off of the `test_execution_id` given to the workload test during its last run. 
+
+While [running a test](https://opensearch.org/docs/latest/benchmark/reference/commands/execute-test/#general-settings), you can also customize where the results are stored, using any combination of the following command flags:
+
+* `--results-file`: When provided a file path, writes the compare results to the file indicated in the path.
+* `--results-format`: Defines the output format for the command line results, either markdown or csv. Default is `markdown`.
+* `--show-in-results`: Determines whether or not to include a comparison to previous runs in the results file. 
+* `--user-tag`: Defines user-specific key-value pairs to be used in metric record as meta information, for example, `intention:baseline-ticket-12345`.
```
Comment (IanHoang) [inline _benchmark/user-guide/understanding-results.md:124]: We can also add how this is useful if storing in external metric stores

File: _benchmark/user-guide/understanding-results.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+
+The following guide gives information about to understand the results of the summary report and what steps to take before running another test on your cluster.
+
+## OpenSearch Benchmark runs
+
+OpenSearch Benchmark runs a series of nightly tests targeting the overall OpenSearch development cluster. These runs can be found on https://opensearch.org/benchmarks. It compares several metrics across different test runs targeting both recent and future versions of OpenSearch.
+
+## Selecting metrics to compare
+
+While an OpenSearch Benchmark summary report provides many metrics related to the performance of your cluster, how to compare and use those metrics depends on your use case. Some users might be interested in the number of documents their can index, while another might be interested in how much latency it takes to query a document. For example, during the OpenSearch Benchmark nightly runs, the OpenSearch teams pulls metrics similar to the following summary report:
```
Comment (IanHoang) [inline _benchmark/user-guide/understanding-results.md:19]: latency or service time

**Review by IanHoang — COMMENTED** [review]
We should stick with summary report (instead of using command line results)

File: _benchmark/user-guide/understanding-results.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
+---
+layout: default
+title: Understanding results
+nav_order: 22
+parent: User guide
+---
+
+
+At the end of each test run, a summary table is produced which includes metrics like service time, throughput, latency and more. These metrics provide insights into how the workload selected performed on a benchmarked OpenSearch cluster.
```
Comment (IanHoang) [inline _benchmark/user-guide/understanding-results.md:9]: a summary results table

Comment (IanHoang) [reply _benchmark/user-guide/understanding-results.md:9]: on an OpenSearch cluster

File: _benchmark/user-guide/understanding-results.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+
+The following guide gives information about to understand the results of the summary report and what steps to take before running another test on your cluster.
```
Comment (IanHoang) [inline _benchmark/user-guide/understanding-results.md:11]: about how to understand the summary results

Comment (IanHoang) [reply _benchmark/user-guide/understanding-results.md:11]: I think we can also remove "what steps to take before running another test on your cluster" as that's not found here.

**Review by IanHoang — CHANGES_REQUESTED** [review]
Left some comments

File: _benchmark/user-guide/understanding-results.md (hunk 8 above)
Comment (Naarcha-AWS) [inline _benchmark/user-guide/understanding-results.md:9]:
```suggestion
At the end of each test run, a summary table is produced which includes metrics like service time, throughput, latency, and more. These metrics provide insights into how the workload selected performed on a benchmarked OpenSearch cluster.
```

File: _benchmark/user-guide/understanding-results.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+
+## OpenSearch Benchmark runs
+
+OpenSearch Benchmark runs a series of nightly tests targeting the overall OpenSearch development cluster. These runs can be found on https://opensearch.org/benchmarks. It compares several metrics across different test runs targeting both recent and future versions of OpenSearch.
+
+## Selecting metrics to compare
+
+While an OpenSearch Benchmark summary report provides many metrics related to the performance of your cluster, how to compare and use those metrics depends on your use case. Some users might be interested in the number of documents their can index, while another might be interested in how much latency it takes to query a document. For example, during the OpenSearch Benchmark nightly runs, the OpenSearch teams pulls metrics similar to the following summary report:
```
Comment (Naarcha-AWS) [inline _benchmark/user-guide/understanding-results.md:19]:
```suggestion
While an OpenSearch Benchmark summary report provides many metrics related to the performance of your cluster, how to compare and use those metrics depends on your use case. Some users might be interested in the number of documents their can index, while another might be interested in the amount of latency or service time it takes for a document to be queried. For example, during the OpenSearch Benchmark nightly runs, the OpenSearch teams pulls metrics similar to the following summary report:
```

File: _benchmark/user-guide/understanding-results.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+Results from OpenSearch Benchmark are stored in two ways, either in-memory or in an external metric store. 
+
+When stored in-memory, results can be found in the `/.benchmark/benchmarks/test_executions/<test_execution_id>` directory. Results are named based off of the `test_execution_id` given to the workload test during its last run. 
+
+While [running a test](https://opensearch.org/docs/latest/benchmark/reference/commands/execute-test/#general-settings), you can also customize where the results are stored, using any combination of the following command flags:
+
+* `--results-file`: When provided a file path, writes the compare results to the file indicated in the path.
+* `--results-format`: Defines the output format for the command line results, either `markdown` or `csv`. Default is `markdown`.
+* `--show-in-results`: Determines whether or not to include a comparison to previous runs in the results file. 
+* `--user-tag`: Defines user-specific key-value pairs to be used in metric record as meta information, for example, `intention:baseline-ticket-12345`.
```
Comment (Naarcha-AWS) [inline _benchmark/user-guide/understanding-results.md:124]:
```suggestion
* `--user-tag`: Defines user-specific key-value pairs to be used in the metric record as meta information, for example, `intention:baseline-ticket-12345`. This is useful when storing metrics and results in an external metric store.
```

File: _benchmark/user-guide/understanding-results.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+
+Results from OpenSearch Benchmark are stored in two ways, either in-memory or in an external metric store. 
+
+When stored in-memory, results can be found in the `/.benchmark/benchmarks/test_executions/<test_execution_id>` directory. Results are named based off of the `test_execution_id` given to the workload test during its last run. 
+
+While [running a test](https://opensearch.org/docs/latest/benchmark/reference/commands/execute-test/#general-settings), you can also customize where the results are stored, using any combination of the following command flags:
+
+* `--results-file`: When provided a file path, writes the compare results to the file indicated in the path.
+* `--results-format`: Defines the output format for the command line results, either `markdown` or `csv`. Default is `markdown`.
+* `--show-in-results`: Determines whether or not to include a comparison to previous runs in the results file.
```
Comment (Naarcha-AWS) [inline _benchmark/user-guide/understanding-results.md:123]:
```suggestion
* `--show-in-results`:  Defines which values are shown in the published summary report, either `available`, `all-percentiles`, or `all`. Default is `available`.
```

File: _benchmark/user-guide/understanding-results.md [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+## Result storage
+
+Results from OpenSearch Benchmark are stored in two ways, either in-memory or in an external metric store. 
+
+When stored in-memory, results can be found in the `/.benchmark/benchmarks/test_executions/<test_execution_id>` directory. Results are named based off of the `test_execution_id` given to the workload test during its last run. 
+
+While [running a test](https://opensearch.org/docs/latest/benchmark/reference/commands/execute-test/#general-settings), you can also customize where the results are stored, using any combination of the following command flags:
+
+* `--results-file`: When provided a file path, writes the summary report to the file indicated in the path.
+* `--results-format`: Defines the output format for the command line results, either `markdown` or `csv`. Default is `markdown`.
```
Comment (IanHoang) [inline _benchmark/user-guide/understanding-results.md:122]: command line results -> summary report

File: _benchmark/user-guide/understanding-results.md [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+|                                                     error rate |                                       term |           0 |      % |
+```
+
+Metrics unique to the cluster begin at the `index` task line. The following two use cases can give you an idea of what metrics might be relevant to you:
+
+- To assess how much load your cluster can handle, the `index` task metrics provide the number of documents ingested during the workload run, as well as the ingestion error rate. 
+- To assess the measurable latency and service time of the queries in the workload, the `match_all` and `term` give both the number of query operations performed per second and the measurable latency of the query, as well as the error rate when running query operations.
+
+
+## Result storage
```
Comment (IanHoang) [inline _benchmark/user-guide/understanding-results.md:113]: SUPER Nit: Result -> Results

File: _benchmark/user-guide/understanding-results.md (hunk 13 above)
Comment (Naarcha-AWS) [inline _benchmark/user-guide/understanding-results.md:122]:
```suggestion
* `--results-format`: Defines the output format for the summary report results, either `markdown` or `csv`. Default is `markdown`.
```

File: _benchmark/user-guide/understanding-results.md (hunk 14 above)
Comment (Naarcha-AWS) [inline _benchmark/user-guide/understanding-results.md:113]:
```suggestion
## Results storage
```

**Review by hdhalter — COMMENTED** [review]
Thanks, Nate! I reviewed the first half and will do the second half later today. I left a few suggestions for reorganizing some of the info and some wording tweaks.

File: _benchmark/user-guide/understanding-results.md [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+|                                                 Min Throughput |                                       term |        16.1 |  ops/s |
+|                                                Mean Throughput |                                       term |        16.1 |  ops/s |
+|                                              Median Throughput |                                       term |        16.1 |  ops/s |
+|                                                 Max Throughput |                                       term |        16.1 |  ops/s |
+|                                       100th percentile latency |                                       term |     131.798 |     ms |
+|                                  100th percentile service time |                                       term |     69.5237 |     ms |
+|                                                     error rate |                                       term |           0 |      % |
+```
+
+Metrics unique to the cluster begin at the `index` task line. The following two use cases can give you an idea of what metrics might be relevant to you:
```
Comment (hdhalter) [inline _benchmark/user-guide/understanding-results.md:107]:
```suggestion
Metrics that are unique to the cluster begin at the `index` task line. The following two use cases are examples of metrics that might be relevant to you:
```

File: _benchmark/user-guide/understanding-results.md [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+
+- To assess how much load your cluster can handle, the `index` task metrics provide the number of documents ingested during the workload run, as well as the ingestion error rate. 
+- To assess the measurable latency and service time of the queries in the workload, the `match_all` and `term` give both the number of query operations performed per second and the measurable latency of the query, as well as the error rate when running query operations.
```
Comment (hdhalter) [inline _benchmark/user-guide/understanding-results.md:110]:
```suggestion
- To assess the measurable latency and service time of the queries in the workload, the `match_all` and `term` tasks provide both the number of query operations performed per second and the measurable latency of the query, as well as the error rate when running query operations.
```

File: _benchmark/user-guide/understanding-results.md [hunk 17]
```diff
@@ -0,0 +1,125 @@
...
+
+
+## Results storage
+
+Results from OpenSearch Benchmark are stored in two ways, either in-memory or in an external metric store.
```
Comment (hdhalter) [inline _benchmark/user-guide/understanding-results.md:115]:
```suggestion
Results from OpenSearch Benchmark are stored in-memory or in an external metric store. 
```

File: _benchmark/user-guide/understanding-results.md [hunk 18]
```diff
@@ -0,0 +1,125 @@
...
+
+- To assess how much load your cluster can handle, the `index` task metrics provide the number of documents ingested during the workload run, as well as the ingestion error rate. 
+- To assess the measurable latency and service time of the queries in the workload, the `match_all` and `term` give both the number of query operations performed per second and the measurable latency of the query, as well as the error rate when running query operations.
+
+
+## Results storage
+
+Results from OpenSearch Benchmark are stored in two ways, either in-memory or in an external metric store. 
+
+When stored in-memory, results can be found in the `/.benchmark/benchmarks/test_executions/<test_execution_id>` directory. Results are named based off of the `test_execution_id` given to the workload test during its last run.
```
Comment (hdhalter) [inline _benchmark/user-guide/understanding-results.md:117]:
```suggestion
When stored in-memory, results are found in the `/.benchmark/benchmarks/test_executions/<test_execution_id>` directory. The name of the result is based on the `test_execution_id` given to the workload test during its most recent run. 
```

File: _benchmark/user-guide/understanding-results.md [hunk 19]
```diff
@@ -0,0 +1,125 @@
...
+- To assess the measurable latency and service time of the queries in the workload, the `match_all` and `term` give both the number of query operations performed per second and the measurable latency of the query, as well as the error rate when running query operations.
+
+
+## Results storage
+
+Results from OpenSearch Benchmark are stored in two ways, either in-memory or in an external metric store. 
+
+When stored in-memory, results can be found in the `/.benchmark/benchmarks/test_executions/<test_execution_id>` directory. Results are named based off of the `test_execution_id` given to the workload test during its last run. 
+
+While [running a test](https://opensearch.org/docs/latest/benchmark/reference/commands/execute-test/#general-settings), you can also customize where the results are stored, using any combination of the following command flags:
```
Comment (hdhalter) [inline _benchmark/user-guide/understanding-results.md:119]:
```suggestion
While [running a test](https://opensearch.org/docs/latest/benchmark/reference/commands/execute-test/#general-settings), you can customize where the results are stored using any combination of the following command flags:
```

**Review by hdhalter — COMMENTED** [review]
HI Nate, here are a few tweaks to the second half.

File: _benchmark/user-guide/understanding-results.md [hunk 20]
```diff
@@ -0,0 +1,125 @@
...
+
+The following guide gives information about how to understand the results of the summary report.
+
+## OpenSearch Benchmark runs
+
+OpenSearch Benchmark runs a series of nightly tests targeting the overall OpenSearch development cluster. These runs can be found on https://opensearch.org/benchmarks. It compares several metrics across different test runs targeting both recent and future versions of OpenSearch.
+
+## Selecting metrics to compare
+
+While an OpenSearch Benchmark summary report provides many metrics related to the performance of your cluster, how to compare and use those metrics depends on your use case. Some users might be interested in the number of documents their can index, while another might be interested in the amount of latency or service time it takes for a document to be queried. For example, during the OpenSearch Benchmark nightly runs, the OpenSearch teams pulls metrics similar to the following summary report:
```
[the remark this answers is not in the document]
Comment (Naarcha-AWS) [reply _benchmark/user-guide/understanding-results.md:19]: Switched to "during a typical workload run."

File: _benchmark/user-guide/understanding-results.md [hunk 21]
```diff
@@ -0,0 +1,125 @@
+---
+layout: default
+title: Understanding results
```
[the remark this answers is not in the document]
Comment (Naarcha-AWS) [reply _benchmark/user-guide/understanding-results.md:3]: "Understanding benchmark results?" What do you think @IanHoang

File: _benchmark/user-guide/understanding-results.md [hunk 22]
```diff
@@ -0,0 +1,125 @@
...
+nav_order: 22
+parent: User guide
+---
+
+
+At the end of each test run, a summary table is produced which includes metrics like service time, throughput, latency, and more. These metrics provide insights into how the workload selected performed on a benchmarked OpenSearch cluster.
```
Comment (Naarcha-AWS) [inline _benchmark/user-guide/understanding-results.md:9]: @hdhalter: Is this better?

```suggestion
At the end of each test run, OpenSearch Benchmark creates summary of test results metrics like service time, throughput, latency, and more. These metrics provide insights into how the workload selected performed on a benchmarked OpenSearch cluster.
```

File: _benchmark/user-guide/understanding-results.md [hunk 23]
```diff
@@ -0,0 +1,125 @@
...
+
+The following guide gives information about how to understand the results of the summary report.
+
+## OpenSearch Benchmark runs
+
+OpenSearch Benchmark runs a series of nightly tests targeting the overall OpenSearch development cluster. These runs can be found on https://opensearch.org/benchmarks. It compares several metrics across different test runs targeting both recent and future versions of OpenSearch.
```
[the remark this answers is not in the document]
Comment (Naarcha-AWS) [reply _benchmark/user-guide/understanding-results.md:15]: This is specific to the link provided in the paragraph. It only applies as an example and not as the standard for summary reports. In the orginal draft, this section was at the bottom and included a disclaimer similar to the following, "Use the nightly benchmark runs as an example of how to present your benchmark results in OpenSearch Dashboards."

File: _benchmark/user-guide/understanding-results.md [hunk 24]
```diff
@@ -0,0 +1,125 @@
...
+
+- To assess how much load your cluster can handle, the `index` task metrics provide the number of documents ingested during the workload run, as well as the ingestion error rate. 
+- To assess the measurable latency and service time of the queries in the workload, the `match_all` and `term` tasks provide both the number of query operations performed per second and the measurable latency of the query, as well as the error rate when running query operations.
+
+
+## Results storage
+
+Results from OpenSearch Benchmark are stored in-memory or in an external metric store. 
+
+When stored in-memory, results can be found in the `/.benchmark/benchmarks/test_executions/<test_execution_id>` directory. Results are named based off of the `test_execution_id` given to the workload test during its last run.
```
Comment (Naarcha-AWS) [inline _benchmark/user-guide/understanding-results.md:117]:
```suggestion
When stored in-memory, results can be found in the `/.benchmark/benchmarks/test_executions/<test_execution_id>` directory. Results are named based off of the `test_execution_id` given to the workload test during its most recent run. 
```

File: _benchmark/user-guide/understanding-results.md (hunk 21 above)
Comment (Naarcha-AWS) [inline _benchmark/user-guide/understanding-results.md:3]:
```suggestion
title: Understanding benchmark results
```

File: _benchmark/user-guide/understanding-results.md [hunk 25]
```diff
@@ -0,0 +1,125 @@
...
+
+At the end of each test run, OpenSearch Benchmark creates a summary of test results metrics like service time, throughput, latency, and more. These metrics provide insights into how the workload selected performed on a benchmarked OpenSearch cluster.
+
+The following guide gives information about how to understand the results of the summary report.
+
+## OpenSearch Benchmark runs
+
+OpenSearch Benchmark runs a series of nightly tests targeting the OpenSearch development cluster. The results of these test runs can be found on https://opensearch.org/benchmarks. The results show the metrics that span different test runs and target both recent and future versions of OpenSearch.
+
+## Selecting metrics to compare
```
[the remark this answers is not in the document]
Comment (Naarcha-AWS) [reply _benchmark/user-guide/understanding-results.md:17]: The section is meant to help guide users towards finding the metrics important for them, so while there isn't a step through, whats important to users will be different depending on their use case. 

Lets get @natebower take on this.

The original section title was ## Important metrics

File: _benchmark/user-guide/understanding-results.md [hunk 26]
```diff
@@ -0,0 +1,125 @@
...
+---
+
+
+At the end of each test run, OpenSearch Benchmark creates a summary of test results metrics like service time, throughput, latency, and more. These metrics provide insights into how the workload selected performed on a benchmarked OpenSearch cluster.
+
+The following guide gives information about how to understand the results of the summary report.
+
+## OpenSearch Benchmark runs
+
+OpenSearch Benchmark runs a series of nightly tests targeting the OpenSearch development cluster. The results of these test runs can be found on https://opensearch.org/benchmarks. The results show the metrics that span different test runs and target both recent and future versions of OpenSearch.
```
Comment (hdhalter) [inline _benchmark/user-guide/understanding-results.md:15]: Can we define 'runs' here if that is the terminology that we are using for the nightly test? OpenSearch Benchmark runs are the nightly tests that target the OS dev cluster..."

File: _benchmark/user-guide/understanding-results.md [hunk 27]
```diff
@@ -0,0 +1,125 @@
...
+layout: default
+title: Understanding benchmark results
+nav_order: 22
+parent: User guide
+---
+
+
+At the end of each test run, OpenSearch Benchmark creates a summary of test results metrics like service time, throughput, latency, and more. These metrics provide insights into how the workload selected performed on a benchmarked OpenSearch cluster.
+
+The following guide gives information about how to understand the results of the summary report.
```
Comment (natebower) [inline _benchmark/user-guide/understanding-results.md:11]:
```suggestion
The following guide provides information about how to understand the results of the summary report.
```

File: _benchmark/user-guide/understanding-results.md (hunk 26 above)
Comment (natebower) [inline _benchmark/user-guide/understanding-results.md:15]:
```suggestion
OpenSearch Benchmark runs a series of nightly tests targeting the OpenSearch development cluster. The results of these test runs can be found at https://opensearch.org/benchmarks. The results display metrics spanning different test runs and target both recent and future versions of OpenSearch.
```

File: _benchmark/user-guide/understanding-results.md [hunk 28]
```diff
@@ -0,0 +1,125 @@
...
+
+## OpenSearch Benchmark runs
+
+OpenSearch Benchmark runs a series of nightly tests targeting the OpenSearch development cluster. The results of these test runs can be found on https://opensearch.org/benchmarks. The results show the metrics that span different test runs and target both recent and future versions of OpenSearch.
+
+## Selecting metrics to compare
+
+While an OpenSearch Benchmark summary report provides metrics related to the performance of your cluster, how to compare and use those metrics depends on your use case. For example, some users might be interested in the number of documents the workload can index, while another might be interested in the amount of latency or service time it takes for a document to be queried. The following example shows a summary report with metrics during a typical workload run:
```
Comment (natebower) [inline _benchmark/user-guide/understanding-results.md:19]:
```suggestion
While an OpenSearch Benchmark summary report provides metrics related to the performance of your cluster, how you compare and use those metrics depends on your use case. For example, some users might be interested in the number of documents the workload can index, while another might be interested in the amount of latency or service time needed for a document to be queried. The following example summary report shows metrics for a typical workload run:
```

File: _benchmark/user-guide/understanding-results.md [hunk 29]
```diff
@@ -0,0 +1,125 @@
...
+|                                                 Min Throughput |                                       term |        16.1 |  ops/s |
+|                                                Mean Throughput |                                       term |        16.1 |  ops/s |
+|                                              Median Throughput |                                       term |        16.1 |  ops/s |
+|                                                 Max Throughput |                                       term |        16.1 |  ops/s |
+|                                       100th percentile latency |                                       term |     131.798 |     ms |
+|                                  100th percentile service time |                                       term |     69.5237 |     ms |
+|                                                     error rate |                                       term |           0 |      % |
+```
+
+Metrics that are unique to the cluster begin at the `index` task line. The following two use cases are examples of metrics that might be relevant to you:
```
Comment (natebower) [inline _benchmark/user-guide/understanding-results.md:107]:
```suggestion
Metrics that are unique to the cluster begin at the `index` task line. The following are examples of metrics that might be relevant to you:
```

File: _benchmark/user-guide/understanding-results.md [hunk 30]
```diff
@@ -0,0 +1,125 @@
...
+
+- To assess how much load your cluster can handle, the `index` task metrics provide the number of documents ingested during the workload run, as well as the ingestion error rate.
```
Comment (natebower) [inline _benchmark/user-guide/understanding-results.md:109]:
```suggestion
- To assess how much load your cluster can handle, the `index` task metrics provide the number of documents ingested during the workload run as well as the ingestion error rate. 
```

File: _benchmark/user-guide/understanding-results.md [hunk 31]
```diff
@@ -0,0 +1,125 @@
...
+|                                                 Max Throughput |                                       term |        16.1 |  ops/s |
+|                                       100th percentile latency |                                       term |     131.798 |     ms |
+|                                  100th percentile service time |                                       term |     69.5237 |     ms |
+|                                                     error rate |                                       term |           0 |      % |
+```
+
+Metrics that are unique to the cluster begin at the `index` task line. The following two use cases are examples of metrics that might be relevant to you:
+
+- To assess how much load your cluster can handle, the `index` task metrics provide the number of documents ingested during the workload run, as well as the ingestion error rate. 
+- To assess the measurable latency and service time of the queries in the workload, the `match_all` and `term` tasks provide both the number of query operations performed per second and the measurable latency of the query, as well as the error rate when running query operations.
```
Comment (natebower) [inline _benchmark/user-guide/understanding-results.md:110]:
```suggestion
- To assess the measurable latency and service time of the queries in the workload, the `match_all` and `term` tasks provide both the number of query operations performed per second and the measurable latency of the query as well as the error rate when running query operations.
```

Comment (natebower) [inline _benchmark/user-guide/understanding-results.md:110]: "...provide the number of query operations performed per second, the measurable query latency, and the query operation error rate"?

File: _benchmark/user-guide/understanding-results.md [hunk 32]
```diff
@@ -0,0 +1,125 @@
...
+
+
+## Results storage
```
Comment (natebower) [inline _benchmark/user-guide/understanding-results.md:113]:
```suggestion
## Storing results
```

File: _benchmark/user-guide/understanding-results.md [hunk 33]
```diff
@@ -0,0 +1,125 @@
...
+
+Results from OpenSearch Benchmark are stored in-memory or in an external metric store.
```
Comment (natebower) [inline _benchmark/user-guide/understanding-results.md:115]:
```suggestion
OpenSearch Benchmark results are stored in-memory or in external storage. 
```

File: _benchmark/user-guide/understanding-results.md [hunk 34]
```diff
@@ -0,0 +1,125 @@
...
+
+- To assess how much load your cluster can handle, the `index` task metrics provide the number of documents ingested during the workload run, as well as the ingestion error rate. 
+- To assess the measurable latency and service time of the queries in the workload, the `match_all` and `term` tasks provide both the number of query operations performed per second and the measurable latency of the query, as well as the error rate when running query operations.
+
+
+## Results storage
+
+Results from OpenSearch Benchmark are stored in-memory or in an external metric store. 
+
+When stored in-memory, results can be found in the `/.benchmark/benchmarks/test_executions/<test_execution_id>` directory. Results are named based off of the `test_execution_id` given to the workload test during its most recent run.
```
Comment (natebower) [inline _benchmark/user-guide/understanding-results.md:117]:
```suggestion
When stored in-memory, results can be found in the `/.benchmark/benchmarks/test_executions/<test_execution_id>` directory. Results are named in accordance with the `test_execution_id` of the most recent workload test. 
```

File: _benchmark/user-guide/understanding-results.md [hunk 35]
```diff
@@ -0,0 +1,125 @@
...
+
+Results from OpenSearch Benchmark are stored in-memory or in an external metric store. 
+
+When stored in-memory, results can be found in the `/.benchmark/benchmarks/test_executions/<test_execution_id>` directory. Results are named based off of the `test_execution_id` given to the workload test during its most recent run. 
+
+While [running a test](https://opensearch.org/docs/latest/benchmark/reference/commands/execute-test/#general-settings), you can customize where the results are stored using any combination of the following command flags:
+
+* `--results-file`: When provided a file path, writes the summary report to the file indicated in the path.
+* `--results-format`: Defines the output format for the summary report results, either `markdown` or `csv`. Default is `markdown`.
+* `--show-in-results`:  Defines which values are shown in the published summary report, either `available`, `all-percentiles`, or `all`. Default is `available`.
```
Comment (natebower) [inline _benchmark/user-guide/understanding-results.md:123]:
```suggestion
* `--show-in-results`: Defines which values are shown in the published summary report, either `available`, `all-percentiles`, or `all`. Default is `available`.
```

File: _benchmark/user-guide/understanding-results.md [hunk 36]
```diff
@@ -0,0 +1,125 @@
...
+* `--user-tag`: Defines user-specific key-value pairs to be used in the metric record as meta information, for example, `intention:baseline-ticket-12345`. This is useful when storing metrics and results in an external metric store.
```
Comment (natebower) [inline _benchmark/user-guide/understanding-results.md:124]:
```suggestion
* `--user-tag`: Defines user-specific key-value pairs used in the metrics record as meta information, for example, `intention:baseline-ticket-12345`. This is useful when storing metrics and results in external storage.
```

File: _benchmark/user-guide/understanding-results.md (hunk 25 above)
[the remark this answers is not in the document]
Comment (natebower) [reply _benchmark/user-guide/understanding-results.md:17]: I like "Summary report metrics".

**Review by natebower — COMMENTED** [review]
@Naarcha-AWS Please see my comments and changes and let me know if you have any questions. Thanks!

File: _benchmark/user-guide/understanding-results.md (hunk 25 above)
Comment (Naarcha-AWS) [inline _benchmark/user-guide/understanding-results.md:17]:
```suggestion
## Summary report metrics
```

File: _benchmark/user-guide/understanding-results.md (hunk 26 above)
Comment (Naarcha-AWS) [inline _benchmark/user-guide/understanding-results.md]:
```suggestion
OpenSearch Benchmark team runs nightly tests targeting the OpenSearch development cluster, comparing throughput and latency numbers for both recent and future versions of OpenSearch. The results of these tests can be found on https://opensearch.org/benchmarks. 
```

Comment (Naarcha-AWS) [reply _benchmark/user-guide/understanding-results.md]: Let me know what you think of this @hdhalter

Comment (hdhalter) [reply _benchmark/user-guide/understanding-results.md]: Yes, good to provide the distinction! Just add 'the'. The OpenSearch Benchmark team runs nightly tests..."

**Pull request merged**